core/src/main/java/org/kin/sendkin/core/store/RecipientContactsRepoImpl.java

```diff
@@ -0,0 +1,171 @@
+package org.kin.sendkin.core.store;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Looper;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.VisibleForTesting;
+
+import org.kin.sendkin.core.model.RecipientContact;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.UUID;
+
+public class RecipientContactsRepoImpl implements RecipientContactsRepo {
+
+    private ArrayList<RecipientContact> contacts;
+    private final RecipientContactsLocalStore store;
+    private final String STORE_NAME = "RecipientContactsRepo";
+    private ContactsListener contactsListener = null;
+    private Handler mainHandler = new Handler(Looper.getMainLooper());
+
+    public RecipientContactsRepoImpl(@NonNull Context context) {
+        store = new RecipientContactsLocalStore(new LocalStore(context, STORE_NAME));
+    }
+
+    @Override
+    public void load() {
```

Comment:
Check if it is really worth using a bg thread only for reading from shared prefs

Reply:
@drEfrati since this is an IO operation reading data from the xml that islocated on the disk - i prefer to keep it on bg thread